catalog/resource_grants.go

```diff
@@ -258,13 +324,14 @@ func ResourceGrants() *schema.Resource {
 		Update: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
 			securable, name := mapping.kv(d)
 			diff := permissionDiffFromRaw(d.GetChange("grant"))
-			return NewPermissionsAPI(ctx, c).updatePermissions(securable, name, diff)
+			return NewPermissionsAPI(ctx, c).replacePermissions(securable, name, diff)
```

Comment:
> Per permissionDiffFromRaw, I think this will remove any permissions that haven't changed.

that's the goal - only to have permissions that are within resource definition and nothing else.

> I assume that d.GetChange only returns the permissions that have changed from old to new.

only in tfstate, not on backend

> I'm not familiar with the code enough to say for sure, but it could even remove the need for having permissionDiffFromRaw in the first place.

let's try that

Reply:
1. You'd have the situation where `GRANT x` has been around, is still around, and is returned by the backend. But because it doesn't show up in `diff`, and does show up in the backend, it will be removed.
2. Right, so something that has been around and is still around isn't included.